aikau/src/main/resources/alfresco/lists/AlfFilteredList.js

```diff
@@ -78,23 +92,107 @@ define(["dojo/_base/declare",
        * @instance
        * @param {object} targetNode The node to hide the children of
        */
-      hideChildren: function alfresco_lists_AlfList__hideChildren(/*jshint unused:false*/targetNode) {
+      hideChildren: function alfresco_lists_AlfFilteredList__hideChildren(/*jshint unused:false*/targetNode) {
          this.inherited(arguments);
          domClass.remove(this.filtersNode, "share-hidden");
       },
 
       /**
+       * We need to make sure any filters in the hash are populated into the dataFilters property
+       * 
+       * @instance
+       * @override
+       * @param {object} payload The publication topic
+       */
+      onHashChange: function alfresco_lists_AlfFilteredList__onHashChange( /*jshint unused:false*/ payload) {
+
+         // Only do this when we are mirroring the filters in the hash
+         if (this.mapHashVarsToPayload) {
+
+            // Initialise the data-filters to be all of the filters we have specified, without values
+            this.dataFilters = array.map(Object.keys(this._filterWidgets), function(filterName){
+               return {
+                  name: filterName
+               };
+            });
+
+            // Filter to only include items currently in the hash and update values
+            var currHash = hashUtils.getHash();
+            this.dataFilters = array.filter(this.dataFilters, function(dataFilter) {
+               dataFilter.value = currHash[dataFilter.name];
+               return !!dataFilter.value;
+            });
+
+            // Update the filter fields and reload the data
+            if (this._readyToLoad) {
+            this._updateFilterFieldsFromHash();
+               this.loadData();
+            }
+         }
+
+         // Call inherited
+         this.inherited(arguments);
+      },
+
+      /**
+       * Widget has started
+       *
+       * @instance
+       * @override
+       */
+      startup: function alfresco_lists_AlfFilteredList__startup() {
+         this.inherited(arguments);
+         this._storeFilterWidgets();
+         this._updateFilterFieldsFromHash();
+      },
+
+      /**
+       * Build a collection of filter widgets as a property on this instance
+       *
+       * @instance
+       */
+      _storeFilterWidgets: function alfresco_lists_AlfFilteredList___storeFilterWidgets() {
+         var childWidgets = registry.findWidgets(this.domNode);
+         this._filterWidgets = {};
+         array.forEach(this.widgetsForFilters, function(filterDef) {
+            var filterName = filterDef.config.name;
+            this._filterWidgets[filterName] = array.filter(childWidgets, function(childWidget) {
+               return childWidget.name === filterName;
```

Comment:
No, because registry.findWidgets() returns an empty array by default, i.e. the array.filter will run without error but have nothing to act upon.